pkg/apis/ceph.rook.io/v1/priorityclasses.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1
+
+import (
+	rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+)
+
```

Comment:
What about mds and rgw?

Reply:
Since mds and rgw have their own CRs, they don't have a need to merge with `all` so i'm assuming they can be used directly from the spec

Reply:
From my perspective, ideal behavior for `mds`, `rgw`, `nfs-ganesha`, or any other daemon should be to get the priority class from their CR if it exists, otherwise to default to `all`. Is that the behavior we have present here? Or does anyone see anything wrong with that?

Reply:
@leseb 
The priority class for rgw and mds can be found under:
https://github.com/d-luu/rook/blob/ceph_priority_classes/pkg/operator/ceph/object/spec.go#L74
https://github.com/d-luu/rook/blob/ceph_priority_classes/pkg/operator/ceph/file/mds/spec.go#L57

@BlaineEXE 
Currently the behavior for `all` is for `mgr`, `mon`, `osd`, `rbdmirror` defined in the CephCluster yaml
I could add the `all` behavior to `mds`,`rgw`,`nfs-ganesha` but would we want to pull it from the CephCluster? the priority classes for those are currently are defined in the their own respective CRD

Reply:
I made the change to the behavior for `mds`, `rgw`, `nfs-ganesha` but will need to test if they will pickup the changes in CephCluster to `all` should a user change the value

Reply:
Merging with `all` is an interesting question since it involves inheriting a setting across CRDs. The `placement` currently does not inherit `all` across the object or file CRs, but we do have a pattern of inheriting the ceph image. I think is acceptable to inherit `all` and could certainly be intuitive for users. 

Instead of delaying this PR longer, it may be better to create a separate PR to handle the inheritance of `all` for placement, annotations, and priority classes consistently.